src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.lib.exception.DataProcessingException;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
```

Comment:
You can do it in one line 
```suggestion
            session = HibernateUtil.getSessionFactory().openSession();
```

Reply:
Then no need in line 16